pytorch_to_returnn/torch/nn/utils/rnn.py

```diff
@@ -1,15 +1,31 @@
 import numpy
+from ..functional import full
 from ..modules.rnn import PackedSequence
 from ..modules.shape import FlattenBatch
 from ...tensor import Tensor
 from ..._C import from_numpy
+from .... import torch
 from ....naming import Naming
 
 
+def _batch_sizes_from_lengths(lengths: Tensor) -> Tensor:
+  indices = torch.arange(torch.max(lengths))  # [T]
+  indices_bc = indices.unsqueeze(0)  # [1,T]
+  l_bc = lengths.unsqueeze(1)  # [B,1]
+  mask = indices_bc < l_bc  # [B,T]
+  batch_sizes = torch.sum(mask.int(), dim=0)  # [T]
+  return batch_sizes
+
+
 def pack_padded_sequence(input: Tensor, lengths, batch_first=False, enforce_sorted=True) -> PackedSequence:
-  batch_sizes = []
-  for frame in range(lengths[0]):
-    batch_sizes.append(sum([x > frame for x in lengths]))
+  batch_sizes = None
+  if isinstance(lengths, Tensor):
```

Comment:
We should not need this here. `_batch_sizes_from_lengths` should work for both cases.